### PR TITLE
Add docs for using a private helm repository in wge-profiles

### PR DESCRIPTION
**Pull request by ranatrk**
Closes https://github.com/weaveworks/weave-gitops-enterprise/issues/2649

**What changed?**
Documentation for using private helm repository to retrieve profiles. SecretSync is to be used to sync the secret.

**Why was this change made?**
Document the steps regarding the ability to use private helm repositories as the current usage assumption is of public repos only.

**How was this change implemented?**
N/A

**How did you validate the change?**
N/A

**Release notes**
N/A

**Documentation Changes**

### CODE REVIEW

**Comment by ranatrk** [comment]
@serboctor @enekofb This is a use case of SecretSync to load profiles from private helm repositories. Do you think this should be tied to the other secret docs, or just keeping it with the profiles is enough?

**Comment by serboctor** [comment]
> @serboctor @enekofb This is a use case of SecretSync to load profiles from private helm repositories. Do you think this should be tied to the other secret docs, or just keeping it with the profiles is enough?
@ranatrk  May be we can link to it from the secrets doc. Or actually any doc that mentions profiles/setting up repos. So we have the policy profile and the eso secrets profile

File: website/docs/cluster-management/getting-started.mdx [hunk 1]
```diff
@@ -175,6 +175,8 @@ Download the profile repository below to your config repository path then commit
   {ProfileRepo}
 </CodeBlock>
 
+For more information about profiles, see [profiles from private helm repositories](./profiles.mdx/#optional-using-a-helm-chart-from-a-remote-publicprivate-repository), [policy profiles](../policy/weave-policy-profile.mdx), and [eso secrets profiles](../secrets/setup-eso.mdx).
```
Comment (enekofb) [inline website/docs/cluster-management/getting-started.mdx:178]: given that the scope of the PR is to extend profiles documentation for the use case where the repo is private, i guess that you could remove 

>[policy profiles](../policy/weave-policy-profile.mdx), and [eso secrets profiles](../secrets/setup-eso.mdx).

as these are two examples of profiles ... or you want them for some other reason?

File: website/docs/cluster-management/profiles.mdx [hunk 2]
```diff
@@ -78,6 +78,54 @@ spec:
...
+In this example a public repo and branch is referenced directly where the helm releases are
+```yaml title="HelmRepository.yaml"
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: HelmRepository
+metadata:
+  name: weaveworks-charts
+  namespace: flux-system
+spec:
+  interval: 1m
+  url: https://raw.githubusercontent.com/weaveworks/weave-gitops-profile-examples/gh-pages
```
Comment (enekofb) [inline website/docs/cluster-management/profiles.mdx:93]: this url might be better as points to the webserver https://weaveworks.github.io/weave-gitops-profile-examples/

File: website/docs/cluster-management/profiles.mdx [hunk 3]
```diff
@@ -78,6 +78,54 @@ spec:
...
+metadata:
+  name: my-dev-secret-syncer
+  namespace: flux-system
+spec:
+  clusterSelector:
+    matchLabels:
+      weave.works/capi: bootstrap
+  secretRef:
+    name: weave-gitops-enterprise-credentials
+  targetNamespace: my-namespace
```
Comment (enekofb) [inline website/docs/cluster-management/profiles.mdx:111]: i could think that 

https://github.com/weaveworks/weave-gitops/pull/3620/files#diff-230071a37d2ecb2a12d9cbc206c44ff1750df80f03b9826dd7854c711ac82991R111

might need to be the same as https://github.com/weaveworks/weave-gitops/pull/3620/files#diff-230071a37d2ecb2a12d9cbc206c44ff1750df80f03b9826dd7854c711ac82991R120

so source controller can access the secret?

**Comment by enekofb** [comment]
@ranatrk  it looks great! before ✅ just wondering how was tested e2e ?

**Review by enekofb — APPROVED** [review]
lgtm, before merging, you might need to rebase / adjust this link

>[WARNING] Docs markdown link couldn't be resolved: (../secrets/bootstraping-secrets.mdx) in "cluster-management/profiles.mdx" for version current

**Comment by ranatrk** [comment]
> lgtm, before merging, you might need to rebase / adjust this link
> 
> > [WARNING] Docs markdown link couldn't be resolved: (../secrets/bootstraping-secrets.mdx) in "cluster-management/profiles.mdx" for version current

@enekofb Thank you for the review! With the latest rebase from main this warning doesn't appear and compilation is successful. Manually testing the link works as well.

**Pull request merged**